### PR TITLE
feat(behavior_path_planner): use efficient lateral distance calculation

### PR DESCRIPTION
**Pull request by purewater0901**
## Description
Use more efficient function to compute lateral distance computation.

The results are 
```
Old Computation time: 29345[ns]
New Computation time: 27284[ns]
```
It almost 10% faster than previous one

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/src/util/lane_change/util.cpp [hunk 1]
```diff
@@ -107,6 +107,18 @@ std::vector<int64_t> replaceWithSortedIds(
...
+  const lanelet::ConstLanelets & lanelet_sequence, const geometry_msgs::msg::Pose & pose)
+{
+  lanelet::ConstLanelet closest_lanelet;
+  lanelet::utils::query::getClosestLanelet(lanelet_sequence, pose, &closest_lanelet);
+  const auto & centerline_2d = lanelet::utils::to2D(closest_lanelet.centerline());
+
+  const auto lanelet_point = lanelet::utils::conversion::toLaneletPoint(pose.position);
+  return lanelet::geometry::signedDistance(
+    centerline_2d, lanelet::utils::to2D(lanelet_point).basicPoint());
+}
```
Comment (zulfaqar-azmi-t4) [inline planning/behavior_path_planner/src/util/lane_change/util.cpp:121]: Can this be moved in `utilities`? It doesn't seem to be lane change specific, so other module might benefit from this function as well.

Comment (purewater0901) [reply planning/behavior_path_planner/src/util/lane_change/util.cpp]: @zulfaqar-azmi-t4 done

**Comment by purewater0901** [comment]
evaluator 882/882

**Comment by purewater0901** [comment]
https://evaluation.tier4.jp/evaluation/reports/f7e90b90-c426-5862-b799-47aa1437b4d8?project_id=prd_jt

**Comment by zulfaqar-azmi-t4** [comment]
Seems like there is some issues when computing the lane change path under curved situation.

This is the result before PR

https://user-images.githubusercontent.com/93502286/224270846-903143ae-e9d1-4b9d-b2d8-9478f1eb06e7.mp4

This is the result after PR

https://user-images.githubusercontent.com/93502286/224271128-43b3cea2-8b03-4e30-9c77-fc3dc8a7dc26.mp4

**Comment by purewater0901** [comment]
@zulfaqar-azmi-t4 

I tried quite a few times, but I couldn't reproduce your bags

https://user-images.githubusercontent.com/43805014/224601023-e7f54dad-4e42-4868-bc11-9bd2fdae2279.mp4

**Comment by zulfaqar-azmi-t4** [comment]
@purewater0901 
Yes, I can't reproduce in my home setup as well.

**Review by zulfaqar-azmi-t4 — APPROVED** [review]

**Pull request merged**